packages/ember-routing/lib/location/history_location.js

```diff
@@ -208,14 +207,14 @@ export default EmberObject.extend({
   onUpdateURL(callback) {
     this._removeEventListener();
 
-    this._popstateHandler = () => {
+    this._popstateHandler = run.bind(this, function() {
       // Ignore initial page load popstate event in Chrome
-      if (!popstateFired) {
-        popstateFired = true;
+      if (!this.__popstateFired) {
+        this.__popstateFired = true;
         if (this.getURL() === this._previousURL) { return; }
       }
       callback(this.getURL());
```

Comment:
I'm surprised this (prior to this PR) doesn't trigger the runloops auto-run assertion, causing peoples tests to fail.

We likely need to figure out why, something smells fishy.